### ISSUE TITLE
Running composer require outputs invalid format for platform-dev

### ISSUE THREAD

**Issue by Dropa** [composer/composer#12163]
As of recent versions (somewhere between 2.7.7 and 2.8.1), `composer diagnose` expects `platform-dev` to be object instead of array.

However, when running `require` e.g. `composer require phpunit/phpunit` it will update `platform-dev` as array.

To reproduce;
```
composer require phpunit/phpunit
composer diagnose
# above fails with 
# Checking composer.lock: FAIL
# platform-dev : Array value found, but an object is required

composer update --lock
composer diagnose
# now it works, because lock file was actually switched to object
```

**Comment by Dropa** [comment]
Turns out `platform-dev` is not only section where this fails. Starting from fresh with `composer init`, after require there are several sections that do fail for the same reason.
```
Checking composer.lock: FAIL
stability-flags : Array value found, but an object is required
platform : Array value found, but an object is required
platform-dev : Array value found, but an object is required
```

**Comment by Seldaek** [comment]
I believe this is fixed by #12149 already, will be in 2.8.2
